### PR TITLE
OCPBUGS-27222: Update CloudPrivateIPConfig definition

### PR DESCRIPTION
**Pull request by dulek**
openshift/api#1727 fixes issue where IPv6 CloudPrivateIPConfigs cannot be created because of wrong name validation. This commit imports this change into the CRD created by CNO when deploying cloud-network-config-controller.

### CODE REVIEW

**Comment by dulek** [comment]
I'm not 100% sure this is the proper way to update API version, let's see.

**Comment by kyrtapz** [comment]
@dulek this should do it:
```
go get github.com/openshift/api@master
go mod tidy && go mod vendor
./hack/update-codegen.sh
```
Please add the commands ran into the commit message.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/openshift/cluster-network-operator
 
-go 1.20
+go 1.21
+
+toolchain go1.21.6
```
Comment (dulek) [inline go.mod:5]: @kyrtapz: Is this allowed? `go mod tidy` wants this.

File: go.sum
Comment (dulek) [inline go.sum:1]: @kyrtapz: And all these changes.

**Comment by dulek** [comment]
/retest-required

Alright, this seems to look good now!

**Comment by dulek** [comment]
Closing in favor of #2246.

**Pull request closed without merging**